### PR TITLE
Add scalbn[f|l]function to libc math

### PR DESCRIPTION
**Pull request by flyingfish89**
## Summary
function:
scalbnf:get a float number of x*2^n
The implementation of the function borrows from musl
## Testing
float number1 = scalbnf(2, 10);
printf("%f\n", number1);
number1 = scalbnf(3, 100000);
printf("%f\n", number1);
number = scalbnf(4, 100);
printf("%f\n", number1);
number = scalbnf(5, 1000);
printf("%f\n", number1);
number = scalbnf(12, -10000000);
printf("%f\n", number1);
number = scalbnf(22, 1090000);
printf("%f\n", number1);
number = scalbnf(2075, 10867564);
printf("%f\n", number1);
number = scalbnf(265, -108766);
printf("%f\n", number1);
number = scalbnf(22, -1320);
printf("%f\n", number1);

### CODE REVIEW

File: libs/libc/math/Make.defs [hunk 1]
```diff
@@ -28,7 +28,7 @@ CSRCS += lib_ldexpf.c lib_logf.c lib_log10f.c lib_log2f.c lib_modff.c
 CSRCS += lib_powf.c lib_sinf.c lib_sinhf.c lib_sqrtf.c lib_tanf.c
 CSRCS += lib_tanhf.c lib_asinhf.c lib_acoshf.c lib_atanhf.c lib_erff.c
 CSRCS += lib_copysignf.c
-
+CSRCS += lib_scalbnf.c
```
Comment (xiaoxiang781216) [inline libs/libc/math/Make.defs:31]: merge to previous line

File: libs/libc/math/lib_scalbnf.c [hunk 2]
```diff
@@ -0,0 +1,76 @@
+/****************************************************************************
+ * libs/libc/math/lib_scalbnf.c
+ * get a double number of x*2^n
+ *
+ * This file is a part of NuttX:
```
Comment (xiaoxiang781216) [inline libs/libc/math/lib_scalbnf.c:5]: correct the copyright

File: libs/libc/math/lib_scalbnf.c [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <math.h>
+#include <stdint.h>
+
+#define FLT_MIN 0x1p-126f
```
Comment (xiaoxiang781216) [inline libs/libc/math/lib_scalbnf.c:20]: add preprocess section

File: libs/libc/math/lib_scalbnf.c [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+/****************************************************************************
+ * Name: scalbnf
+ *
+ * Description:
+ *   get a float number of x*2^n
+ *
+ ****************************************************************************/
+
+float scalbnf(float x, int n)
```
Comment (xiaoxiang781216) [inline libs/libc/math/lib_scalbnf.c:33]: what about scalbn/scalbnl

**Comment by xiaoxiang781216** [comment]
please rebase to the last master to fix macOS ci break.

**Comment by xiaoxiang781216** [comment]
Please rebase to the last mater to fix ci broken.

File: libs/libc/math/lib_scalbnf.c (hunk 3 above)
Comment (hartmannathan) [inline libs/libc/math/lib_scalbnf.c:20]:
```suggestion
#define FLT_MIN (0x1p-126f)
```

File: libs/libc/math/lib_scalbnf.c [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+          n -= 127;
+          if (n > 127)
+            {
+              n = 127;
+            }
+        }
+    }
+  else if (n < -126)
+    {
+      y *= FLT_MIN * 0x1p24f;
```
Comment (hartmannathan) [inline libs/libc/math/lib_scalbnf.c:59]: Note that if the above suggestion is not committed, this reads as:

y *= 0x1p - (126f * 0x1p24f);

but if the above suggestion is committed, this reads as:

y *= (0x1p - 126f) * 0x1p24f;

tl;dr: Committing the above suggestion will change the outcome here. Is there a mistake somewhere?

File: libs/libc/math/lib_scalbnf.c [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+      n += 126 - 24;
+      if (n < -126)
+        {
+          y *= FLT_MIN * 0x1p24f;
```
Comment (hartmannathan) [inline libs/libc/math/lib_scalbnf.c:63]: Same note as above.

**Review by xiaoxiang781216 — APPROVED** [review]

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by xiaoxiang781216** [comment]
let's ignore temp ci failure(Documentation and macOS).

**Comment by acassis** [comment]
@xiaoxiang781216 @flyingfish89 @jerpelea since this file is based on musl and uses BSD license, its should include an entry in the file NOTICE. BTW, the commit msg "1" should be avoided.

**Comment by xiaoxiang781216** [comment]
@flyingfish89 please add musl copyright to https://github.com/apache/nuttx/blob/master/LICENSE. Please update your commit message carefully in the new patch.

**Comment by hartmannathan** [comment]
@acassis @xiaoxiang781216 @flyingfish89

Unfortunately the merged commit message cannot be changed without messing up everyone's forks/clones (unless someone knows a git trick that I don't know about).

Fortunately the new PR to add copyright/NOTICE can include a new commit message.

For example:

```
libc/math: Add copyright/NOTICE for scalbnf()

The scalbnf() implementation from musl C library was added to NuttX in commit 035656661e2c626e337f0e85ea99d742c9d85c62 (GitHub PR 7850).
```

or something like that.

Thanks!

**Comment by acassis** [comment]
Yes, we cannot rewind the git because git could break things for everybody that already synced with the upstream. When we do "git push -f" in our fork it doesn't rewind the upstream branch because the PR area is like a hidden branch.

**Comment by xiaoxiang781216** [comment]
Here is the patch to add musl copyright: https://github.com/apache/nuttx/pull/7963
